### DISCUSSION TITLE
Try the new experimental new minimap on the Commit Graph

### DISCUSSION THREAD

**Discussion by eamodio** [gitkraken/vscode-gitlens#2477] [Labs]
We're excited about a new experimental "minimap" feature that just landed in the pre-release (or insiders) version of GitLens on the Commit Graph.

![image](https://user-images.githubusercontent.com/641685/215598212-9796a6fb-2f85-4fc8-86b8-a28d2b86a41b.png)

The idea behind this minimap is to provide a whole new dimension to the Commit Graph. Quickly see the activity of the repo, see HEAD/upstream, branches (local and remote), and tags and easily jump to them. Search results are also shown on the minimap to give you a better idea of where those changes came in.

While there isn't a "legend" or anything just yet -- a quick overview. The green line(s) (markers) denote HEAD & upstream (these will ultimately be a different color), the "highlighted" region is the area of the Commit Graph that is scrolled into view, and the short lines (markers) at the bottom are: small blue = tags, small orange = remote branches, longer orange local branches.

This is still very early and are currently still playing with colors, additional features/discovery, etc. We really want to hear your feedback -- what do you think, is this useful, what does it have that you like, what is it missing, etc.

You can try this out today by switching to the pre-release edition of GitLens, by clicking on the "Switch to Pre-Release version of this extension" from the Extensions view.

![image](https://user-images.githubusercontent.com/641685/199397587-6948b1bc-3019-4d85-a6d5-332fea5d8f78.png)

**Comment by tjx666** [comment]
Why the date from `left to right` is `new to old`?

**Reply by eamodio** [reply]
We actually had a lot of discussions on that internally. We ultimately landed on trying to keep the "area of focus" in the top left -- so the top of the Graph is the most recent, and the left of the minimap is most recent. Also, as you scroll down the Graph, you will see the visible region "scroll" along with you, and it felt natural to have that move from left to right as the Graph is scrolling down.

**Comment by ooliver1** [comment]
![image](https://github.com/gitkraken/vscode-gitlens/assets/34910574/9249bc5f-8f5c-4a02-9623-6cf0d919e505)

any idea as to why this does not work with my (admittedly kinda outdated) theme?

**Reply by eamodio** [reply]
What isn't working? is that purple block in the middle there?

**Reply by ooliver1** [reply]
Yes, in your image it shows that as a transparent highlight, not a full on colour rectangle blocking the whole point of it highlighting.

**Reply by eamodio** [reply]
Can you point us to the theme you are using?

**Reply by ooliver1** [reply]
https://github.com/ooliver1/vsCode-dark-theme
